R/summarystatsbayesianztest.R

```diff
@@ -0,0 +1,410 @@
+      "lessThanTestValue"    = "smaller"
+    ),
+    xName      = if (options[["dataType"]] == "esAndCiLog") gettext("exp(Effect size)") else gettext("Effect size")
+  )
+
+  priorPosteriorPlot[["plotObject"]] <- plot
+
+  return()
+}
+.bayesianZTestRobustnessPlot     <- function(jaspResults, options) {
+
+  if (!is.null(jaspResults[["robustnessPlot"]]))
+    return()
+
+  robustnessPlot <- createJaspPlot(title = gettext("Bayes Factor Robustness Plot"), width = 550, height = 450)
+  robustnessPlot$position <- 3
+  robustnessPlot$dependOn(c(.bayesianZTestsDependencies, c("plotBayesFactorRobustness" ,"robustnessPriorMeanMin", "robustnessPriorMeanMax", "robustnessPriorSdMin", "robustnessPriorSdMax", "plotBayesFactorRobustnessContours", "plotBayesFactorRobustnessContoursValues")))
+  jaspResults[["robustnessPlot"]] <- robustnessPlot
+
+  # specify contour breaks
+  formatBF <- function(BF) ifelse(BF < 1, paste0("1/", round(1/BF, 2)), as.character(round(BF, 2)))
+  if (options[["plotBayesFactorRobustnessContours"]]) {
+    bfBreaks <- strsplit(options[["plotBayesFactorRobustnessContoursValues"]], ",")[[1]]
+    bfBreaks <- sapply(bfBreaks, function(bfBreak) eval(parse(text = bfBreak)))
+    if (anyNA(bfBreaks)) {
+      robustnessPlot$setError(gettext("Bayes factor contours breaks could not be parsed into numbers."))
+      return()
+    }
+  } else {
+    bfBreaks <- c(1/100, 1/30, 1/10, 1/3, 1, 3, 10, 30, 100)
+  }
+  bfLims   <- range(bfBreaks) * c(0.95, 1.05)
+  bfLabs   <- formatBF(bfBreaks)
+
+  # compute BF grid
+  data      <- jaspResults[["data"]][["object"]]
+  priorM    <- seq(options[["robustnessPriorMeanMin"]], options[["robustnessPriorMeanMax"]], length.out = 100)
+  priorSD   <- seq(options[["robustnessPriorSdMin"]],   options[["robustnessPriorSdMax"]],   length.out = 100)
+  priorSD   <- priorSD[priorSD > 0.001] # in order to keep contours from sliding into the edges
+  priorGrid <- expand.grid(mean = priorM, sd = priorSD)
+  alt       <- switch(
+    options[["hypothesis"]],
+    "notEqualToTestValue"  = "two.sided",
+    "greaterThanTestValue" = "greater",
+    "lessThanTestValue"    = "less"
+  )
+  priorGrid$bf <- .bayesianZTestsComputeBf10(x = data[["es"]], se = data[["se"]], mu = priorGrid$mean, tau = priorGrid$sd, alternative = alt)
+  priorGrid$bf[priorGrid$bf < bfLims[1]] <- bfLims[1]
+  priorGrid$bf[priorGrid$bf > bfLims[2]] <- bfLims[2]
+
+  # get plot
+  meanBreaks <- jaspGraphs::getPrettyAxisBreaks(range(priorM))
+  sdBreaks   <- jaspGraphs::getPrettyAxisBreaks(range(priorSD))
+  contours   <- bfBreaks
+
+  plot <- ggplot2::ggplot(
+    data    = priorGrid,
+    mapping = ggplot2::aes(x = sd, y = mean)
+    ) +
+    ggplot2::geom_raster(
+      mapping = ggplot2::aes(fill = bf),
+      alpha   = 0.8
+    ) +
+    ggplot2::geom_contour(
+      mapping = ggplot2::aes(z = bf),
+      breaks  = contours,
+      color   = "black", lty = 2, alpha = 0.3
+    ) +
+    metR::geom_text_contour(
```

Comment:
[`metR`](https://github.com/eliocamp/metR) also adds a bootload of dependencies (See https://github.com/eliocamp/metR/blob/master/DESCRIPTION). @FBartos is it really important to show the numbers on top of the contour lines?

Reply:
they look really nice though :(

I was looking whether I could copy some of the functions to cut on dependencies, but it would be 300+ lines of code at least.... :/

Reply:
Yeah, I just tried and it's a ton of code (I gave up after 500+ lines of code). All the more reason not to do this IMO. If you really want this though, I suggest you cut it out of metR and make a PR to jaspGraphs.

Reply:
ok, I removed it for now :(